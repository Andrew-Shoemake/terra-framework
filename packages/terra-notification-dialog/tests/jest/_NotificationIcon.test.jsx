import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import NotificationIcon from '../../src/_NotificationIcon';

describe('Notification Icon', () => {
  it('shallow renders alert icon', () => {
    const icon = shallowWithIntl(
      <NotificationIcon variant="alert" />,
    );

    expect(icon).toMatchSnapshot();
  });

  it('shallow renders error icon', () => {
    const icon = shallowWithIntl(
      <NotificationIcon variant="error" />,
    );

    expect(icon).toMatchSnapshot();
  });

  it('shallow renders warning icon', () => {
    const icon = shallowWithIntl(
      <NotificationIcon variant="warning" />,
    );

    expect(icon).toMatchSnapshot();
  });

  it('shallow renders info icon', () => {
    const icon = shallowWithIntl(
      <NotificationIcon variant="info" />,
    );

    expect(icon).toMatchSnapshot();
  });

  it('shallow renders custom icon container', () => {
    const icon = shallowWithIntl(
      <NotificationIcon variant="custom" />,
    );

    expect(icon).toMatchSnapshot();
  });

  it('shallow renders custom icon container', () => {
    const icon = shallowWithIntl(
      <NotificationIcon
        variant="custom"
        iconClassName="my-icon-class"
      />,
    );

    expect(icon.find('.my-icon-class')).toHaveLength(1);
    expect(icon).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'orion-fusion-theme',
      });

    const icon = shallowWithIntl(
      <NotificationIcon variant="alert" />,
    );

    expect(icon).toMatchSnapshot();
  });
});
