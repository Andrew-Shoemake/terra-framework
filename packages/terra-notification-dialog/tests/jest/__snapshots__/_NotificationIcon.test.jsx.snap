// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Icon correctly applies the theme context className 1`] = `
<div
  className="icon-container orion-fusion-theme"
>
  <span
    className="icon alert-icon"
  />
</div>
`;

exports[`Notification Icon shallow renders alert icon 1`] = `
<div
  className="icon-container"
>
  <span
    className="icon alert-icon"
  />
</div>
`;

exports[`Notification Icon shallow renders custom icon container 1`] = `
<div
  className="icon-container"
>
  <span
    className="icon custom-icon undefined"
  />
</div>
`;

exports[`Notification Icon shallow renders custom icon container 2`] = `
<div
  className="icon-container"
>
  <span
    className="icon custom-icon my-icon-class"
  />
</div>
`;

exports[`Notification Icon shallow renders error icon 1`] = `
<div
  className="icon-container"
>
  <span
    className="icon error-icon"
  />
</div>
`;

exports[`Notification Icon shallow renders info icon 1`] = `
<div
  className="icon-container"
>
  <span
    className="icon info-icon"
  />
</div>
`;

exports[`Notification Icon shallow renders warning icon 1`] = `
<div
  className="icon-container"
>
  <span
    className="icon warning-icon"
  />
</div>
`;
